alibi_detect/saving/schemas.py

```diff
@@ -53,6 +52,20 @@ def coerce_int2list(value: int) -> List[int]:
         return value
 
 
+def validate_model(model: Callable, values: dict) -> Callable:
+    """Validator to check the model is compatible with the given backend"""
+    backend = values['backend']
+    if backend == 'tensorflow':
+        assert isinstance(model, SupportedModels_tf)
+    elif backend == 'pytorch':
+        assert isinstance(model, SupportedModels_torch)
+    elif backend == 'sklearn':
+        assert isinstance(model, SupportedModels_sklearn)
+    return model
```

Comment:
I'm not too familiar with `pydantic` validators, but shouldn't the `assert` statements be replaced with raising errors?

Reply:
Answering as I implemented this: 

I copied in `assert`'s from the [`pydantic` docs](https://pydantic-docs.helpmanual.io/usage/validators/) without thinking about it enough. You are right though, in their docs they point out "If you make use of assert statements, keep in mind that running Python with the [-O optimization flag](https://docs.python.org/3/using/cmdline.html#cmdoption-o) disables assert statements, and validators will stop working.". To avoid commit clashes would you be able to change these to `raise ValueError(...)` please @mauicv ?